src/lm_buddy/jobs/evaluation/prometheus.py

```diff
@@ -107,25 +108,23 @@ def run_eval(config: PrometheusJobConfig) -> Path:
     # Instantiate OpenAI client to speak with the vLLM endpoint
     client = OpenAI(base_url=config.prometheus.inference.base_url)
 
-    # load dataset from W&B artifact
+    # Load dataset from W&B artifact
     hf_loader = HuggingFaceAssetLoader()
     dataset = hf_loader.load_dataset(config.dataset)
     if config.dataset.prompt_template is not None:
         dataset = format_dataset_with_prompt(
             dataset, config.dataset.prompt_template, config.dataset.text_field
         )
 
-    # get the tokenizer
+    # Get the tokenizer
     tokenizer_config = AutoTokenizerConfig(path=config.prometheus.inference.engine)
     tokenizer = hf_loader.load_pretrained_tokenizer(tokenizer_config)
 
-    # enable / disable tqdm
+    # Enable / disable tqdm
     dataset_iterable = tqdm(dataset) if config.evaluation.enable_tqdm else dataset
 
-    # open the output file for writing and iterate on samples
-    tracking_name = config.tracking.name if config.tracking is not None else "output.json"
-    output_fname = Path(config.evaluation.output_folder) / tracking_name
-    with output_fname.open("w") as file:
+    # Generator that iterates over samples and yields new rows with the prometheus outputs
```

Comment:
@aittalam I've opted to implement the `Dataset.from_generator` trick here to avoid having to write JSON to disk before generating the final dataset.

Reply:
Thanks! It looks like `Dataset.from_generator` [saves the output incrementally](https://huggingface.co/docs/datasets/create_dataset#from-local-files) and then performs memory-mapping. This sounds good as what I was afraid of was what happened if the process broke right before the end (these evals can take hours even on GPUs). As mostly a note to self, we should simulate an issue and see if/how data can be recovered from an interrupted run.

Reply:
Also, we can likely use the batched map and parallelization tools from Datasets to speed up iteration over large datasets.

Reply:
Let's come back to this in a follow-up if we so desire. Easy fixes using `from_generator`, either way.